yaksh/views.py

```diff
@@ -1196,36 +1198,27 @@ def add_teacher(request, course_id):
         raise Http404('You are not allowed to view this page!')
 
     context = {}
-    course = get_object_or_404(Course, creator=user, pk=course_id)
+    course = get_object_or_404(Course, Q(creator=user)|Q(teachers=user), pk=course_id)
     context['course'] = course
 
     if request.method == 'POST':
         teacher_ids = request.POST.getlist('check')
         teachers = User.objects.filter(id__in=teacher_ids)
-        add_to_group(teachers)
-        course.add_teachers(*teachers)
-        context['status'] = True
-        context['teachers_added'] = teachers
+        if not course.creator in teachers:
```

Comment:
Alternative:
In search teacher query we can the course creator.
like: query.exclude(Q(id=course.creator.id)
So you don't need to check the condition neither you need to use any alert in the template.